.github/workflows/pull_request.yml

```diff
@@ -2,7 +2,7 @@ name: code quality checks and unit tests
 
 on:
   pull_request:
-
+  workflow_dispatch:
```

Comment:
Is there a reason to add this ?

Currently tests are not launched automatically on thi PR because you are not yet trusted by github on this repository (first contribution).

Reply:
Do not hesitate to launch tests locally with make tests.